lib/ai/model/tool.go

```diff
@@ -109,18 +134,281 @@ func (*commandExecutionTool) parseInput(input string) (*commandExecutionToolInpu
+	// This is stubbed because accessRequestCreateTool is handled specially.
+	// This is because execution of this tool breaks the loop and returns a suggestion UI prompt.
+	// It is still handled as a tool because testing has shown that the LLM behaves better when it is treated as a tool.
+	//
+	// In addition, treating it as a Tool interface item simplifies the display and prompt assembly logic significantly.
+	return "", trace.NotImplemented("not implemented")
+}
+
+func (*accessRequestCreateTool) parseInput(input string) (*AccessRequest, error) {
+	output, err := parseJSONFromModel[AccessRequest](input)
+	if err != nil {
+		return nil, err
+	}
+
+	if output.Reason == "" {
+		return nil, &invalidOutputError{
+			coarse: "access request create: missing reason",
+			detail: "a reason must be specified for the access request",
+		}
+	}
+
+	if len(output.Roles) == 0 && len(output.Resources) == 0 {
+		return nil, &invalidOutputError{
+			coarse: "access request create: no requested roles or resources",
+			detail: "an access request must be for one or more roles OR one or more resources",
+		}
+	}
+
+	for i, resource := range output.Resources {
+		if resource.Type == "" {
+			return nil, &invalidOutputError{
+				coarse: "access request create: missing type at index " + strconv.Itoa(i),
+				detail: "a type must be provided for each resource",
+			}
+		}
+
+		if resource.Name == "" {
+			return nil, &invalidOutputError{
+				coarse: "access request create: missing name at index " + strconv.Itoa(i),
+				detail: "a name must be provided for each resource",
+			}
+		}
+
+		if resource.Type == types.KindNode {
+			if _, err := uuid.Parse(resource.Name); err != nil {
+				return nil, &invalidOutputError{
+					coarse: "access request create: invalid name at index " + strconv.Itoa(i),
+					detail: "a name must be a valid UUID",
+				}
+			}
+		}
+	}
+
+	return &output, nil
+}
+
+type accessRequestsListTool struct{}
+
+func (*accessRequestsListTool) Name() string {
+	return "List Access Requests"
+}
+
+func (*accessRequestsListTool) Description() string {
+	return "List all access requests that the user has access to."
+}
+
+func (*accessRequestsListTool) Run(ctx context.Context, toolCtx *ToolContext, input string) (string, error) {
+	requests, err := toolCtx.GetAccessRequests(ctx, types.AccessRequestFilter{
+		User: toolCtx.User,
+	})
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	items := make([]accessRequestLLMItem, 0, len(requests))
+	for _, request := range requests {
+		items = append(items, accessRequestLLMItem{
+			Roles:              request.GetRoles(),
+			RequestReason:      request.GetRequestReason(),
+			SuggestedReviewers: request.GetSuggestedReviewers(),
+			State:              request.GetState().String(),
+			ResolveReason:      request.GetResolveReason(),
+			Created:            request.GetCreationTime().Format(time.RFC1123),
```

Comment:
Assist (and most of the Teleport) is using `RFC3339`. Any reason why there a different format is needed?

Reply:
Nope, swapped.